content/public/browser/web_contents.h

```diff
@@ -233,13 +233,9 @@ class WebContents : public PageNavigator, public base::SupportsUserData {
     // If non-null then this WebContents will be hosted by a BrowserPlugin.
     raw_ptr<BrowserPluginGuestDelegate> guest_delegate = nullptr;
 
-    // If non-null then this WebContents will be configured to be hosted
-    // by a different one via the Secure Embed mechanism. The delegate
-    // must outlive this WebContents.
-    // TODO(secure-embed): There needs to be a way of updating this for when
-    // tabs are moved between windows (including potentially an in-between
-    // windows detached tab state).
-    raw_ptr<SecureEmbedDelegate> secure_embed_delegate = nullptr;
+    // If non-null then this WebContents will be configured to be hosted inside
+    // `secure_embed_embedder` via the Secure Embed mechanism.
+    raw_ptr<WebContents> secure_embed_embedder = nullptr;
```

Comment:
This is a subtle but important lifetime requirement change - previously at inner WebContents creation time, the embedder WebContents can be nullptr  (i.e. `SecureEmbedDelegate::GetEmbedderWebContents()` can return nullptr), now the `secure_embed_embedder` must be alive.

I think it is valuable to allow the inner WebContents to be "detached". When detached, the embedder WebContents is nullptr. Can we maybe 1. replaces this parameter with a boolean, e.g. `is_secure_embed_guest` and 2. let `SecureEmbedConnector::Delegate` be the source of truth for the embedder WebContents?

Eventually, I envision we want to switch a WebContents between guest and non-guest, similar to how GuestContents works today, but that's for another PR.